### PR TITLE
Merging dev/release/2.0 into release/2.0.0

### PR DESCRIPTION
**Pull request by tannergooding**
FYI. @dotnet/dotnet-cli

### CODE REVIEW

File: build/sdk-test-env.bat [hunk 1]
```diff
@@ -1,21 +0,0 @@
-@echo off
```
Comment (nguerrera) [inline build/sdk-test-env.bat]: I think my comment about these still being useful was lost in the original PR. Can we please fix up the paths and retain these?

Comment (tannergooding) [reply build/sdk-test-env.bat]: Added a `-dogfood` option and an explicit `build\dogfood.bat` script

File: build/dogfood.cmd [hunk 2]
```diff
@@ -0,0 +1,3 @@
+@echo off
+powershell -NoLogo -NoProfile -ExecutionPolicy ByPass -NoExit %~dp0Build.ps1 -dogfood %*
+exit /b %ErrorLevel%
```
Comment (nguerrera) [inline build/dogfood.cmd]: Will I actually get those environment variables set up for me if I do dogfood.cmd?

Comment (nguerrera) [reply build/dogfood.cmd]: Seems like they'd be trapped in the powershell process

File: build/build.sh [hunk 3]
```diff
@@ -111,6 +117,17 @@ function InstallDotNetCli {
...
 
+  if $dogfood
+  then
+    export SDK_REPO_ROOT="$RepoRoot"
+    export SDK_CLI_VERSION="$DotNetCliVersion"
+    export MSBuildSDKsPath="$ArtifactsConfigurationDir/bin/Sdks"
+    export DOTNET_MSBUILD_SDK_RESOLVER_SDKS_DIR="$MSBuildSDKsPath"
+    export NETCoreSdkBundledVersionsProps="$DotNetRoot/sdk/$DotNetCliVersion/Microsoft.NETCoreSdk.BundledVersions.props"
+    export CustomAfterMicrosoftCommonTargets="$MSBuildSDKsPath/Microsoft.NET.Build.Extensions/msbuildExtensions-ver/Microsoft.Common.Targets/ImportAfter/Microsoft.NET.Build.Extensions.targets"
+    export MicrosoftNETBuildExtensionsTargets="$CustomAfterMicrosoftCommonTargets"
```
Comment (nguerrera) [inline build/build.sh]: I want dotnet that was downloaded on my PATH too in this env.

File: build/dogfood.cmd (hunk 2 above)
Comment (tannergooding) [reply build/dogfood.cmd]: `-NoExit` preserves the powershell context when it leaves

File: build/build.sh (hunk 3 above)
Comment (tannergooding) [reply build/build.sh]: Its added to the path already down below

File: build/dogfood.cmd (hunk 2 above)
Comment (nguerrera) [reply build/dogfood.cmd]: I see.

**Review by nguerrera — APPROVED** [review]

**Pull request merged**